system-addon/lib/ActivityStream.jsm

```diff
@@ -4,113 +4,69 @@
-  "resource://activity-stream/lib/PrefsFeed.jsm");
-
-XPCOMUtils.defineLazyModuleGetter(this, "TelemetryFeed",
-  "resource://activity-stream/lib/TelemetryFeed.jsm");
-
-XPCOMUtils.defineLazyModuleGetter(this, "TopSitesFeed",
-  "resource://activity-stream/lib/TopSitesFeed.jsm");
-
-const PREFS_CONFIG = [
-  {
-    name: "default.sites",
+const PREFS_CONFIG = new Map([
+  ["default.sites", {
     title: "Comma-separated list of default top sites to fill in behind visited sites",
     value: "https://www.facebook.com/,https://www.youtube.com/,https://www.amazon.com/,https://www.yahoo.com/,https://www.ebay.com/,https://twitter.com/"
-  },
-  // When you add a feed pref here:
-  // 1. The pref should be prefixed with "feeds."
-  // 2. The init property should be a function that instantiates your Feed
-  // 3. You should use XPCOMUtils.defineLazyModuleGetter to import the Feed,
-  //    so it isn't loaded until the feed is enabled.
-  {
-    name: "feeds.localization",
-    title: "Initialize strings and detect locale for Activity Stream",
-    value: true,
-    init: () => new LocalizationFeed()
-  },
-  {
-    name: "feeds.newtabinit",
-    title: "Sends a copy of the state to each new tab that is opened",
-    value: true,
-    init: () => new NewTabInit()
-  },
-  {
-    name: "feeds.places",
-    title: "Listens for and relays various Places-related events",
-    value: true,
-    init: () => new PlacesFeed()
-  },
-  {
-    name: "feeds.prefs",
-    title: "Preferences",
-    value: true,
-    init: () => new PrefsFeed(PREFS_CONFIG.map(pref => pref.name))
-  },
-  {
-    name: "feeds.telemetry",
-    title: "Relays telemetry-related actions to TelemetrySender",
-    value: true,
-    init: () => new TelemetryFeed()
-  },
-  {
-    name: "feeds.topsites",
-    title: "Queries places and gets metadata for Top Sites section",
-    value: true,
-    init: () => new TopSitesFeed()
-  },
-  {
-    name: "showSearch",
+  }],
+  ["showSearch", {
     title: "Show the Search bar on the New Tab page",
     value: true
-  },
-  {
-    name: "showTopSites",
+  }],
+  ["showTopSites", {
     title: "Show the Top Sites section on the New Tab page",
     value: true
-  },
-  {
-    name: "telemetry",
+  }],
+  ["telemetry", {
     title: "Enable system error and usage data collection",
     value: false
-  },
-  {
-    name: "telemetry.log",
+  }],
+  ["telemetry.log", {
     title: "Log telemetry events in the console",
     value: false
-  },
-  {
-    name: "telemetry.ping.endpoint",
+  }],
+  ["telemetry.ping.endpoint", {
     title: "Telemetry server endpoint",
     value: "https://tiles.services.mozilla.com/v3/links/activity-stream"
-  }
-];
+  }]
+]);
 
-const feeds = {};
-for (const pref of PREFS_CONFIG) {
-  if (pref.name.match(/^feeds\./)) {
-    feeds[pref.name] = pref.init;
-  }
+const FEEDS_CONFIG = new Map();
+for (const config of [
+  ["localization", "Initialize strings and detect locale for Activity Stream",
```

Comment:
Turns out webpack only expands `[,,,] = ` with the helper function but `{,,,} =` gets expanded to multiple assignments without helper! 👍